0000-zmix/README.md

```diff
@@ -0,0 +1,200 @@
+that have the flexibility and motivation to handle various ZKPs or describe them
+in a generic way. This RFC discusses the data and APIs that are required.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+ZMix will handle three pieces of data for generating and verifying proofs.
+
+## Proof Spec
+
+A document containing the public cryptographic material and descriptive metadata.
+It describes the subproofs that will be used in the overall proof.
+Subproofs are the type of proof(s) to generate and any public data necessary during proof generation.
+For example, anonymous credentials uses the following public data: the schema that describes the credential attributes and an issuers related information like public cryptographic keys.
+Proof specs should be capable of being generated deterministically by any party.
+Used in combination with either a **witness** to generate a proof, or a **proof** to be validated.
+
+## Witness
+
+A document containing the Prover's private data. This includes: private keys, secrets, attribute values, hash pre-images.
+
+## Proof
+
+Cryptographic token(s) that is the proof to be verified.
+
+## Out of scope
+
+Initially, ZMix will not handle creating **proof specs**.
+In a ZKP protocol, a relying party will make a **proof request** for Provers to fulfill.
+A **proof request** contains non-cryptographic information like which credential issuers it will accept,
+how votes will be submitted, and subscriptions are checked. A Prover may not accept the **proof request** as is and could make
+a counter proposal in the form of a **proof offer**. Once a **proof request** is agreed upon, the Prover will have to choose
+how to fulfill the proof. Prover will have to acquire any missing information not already on hand. Prover may also choose between
+multiple credentials that can fulfill the proof. The choices were made will need to be communicated to the verifier as a **proof resolution**–the non-cryptographic
+description about the proof. The **proof resolution** and **proof request** can be combined with the public data to form the **proof spec**.
+Exactly where the public cryptographic material is stored and how it is retrieved is up to end consumers.
+This could be handled later.
+
+## Example diagram
+
+![flow](zmix_flow.png)
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+There are six ZKPs that ZMix can do:
+
+1. *Commitments* - Proof of knowledge of a secret value
+    - Required public parameters: generators, modulus/curve
+1. *Scoped Commitments* - Like *commitments* but restricted to a specific context.
+    - Required public parameters: Hash, context label, map to resulting generator, modulus
+1. *Signature Proofs of Knowledge* - Instead of revealing the signature, prove knowledge of it, allows proving knowledge of messages included in the signature. Messages can be hidden or revealed.
```

Comment:
This module is not a signature proof of knowledge, that's essentially the whole output of zmix. This component proves that you have a signature on (part of) the witness.